File: generate.py
import requests
from os import getenv
import json
import random

API_TOKEN = getenv("HF_API_TOKEN")
models = [
    "runwayml/stable-diffusion-v1-5",
    "CompVis/stable-diffusion-v1-4",
    "stabilityai/stable-diffusion-xl-base-1.0",
    "stabilityai/stable-diffusion-2-1",
    "prompthero/openjourney",
    "prompthero/openjourney-v4",
]

random_model = random.choice(models)
API_URL = f"https://api-inference.huggingface.co/models/{random_model}"
headers = {"Authorization": f"Bearer {API_TOKEN}"}

def query(payload):
    response = requests.post(API_URL, headers=headers, json=payload)
    if response.status_code == 200:
        with open("wallpaper.png", "wb") as f:
            f.write(response.content)

# Load prompt list
with open("prompts.json") as fp:
  prompts = json.load(fp)

# Get random index
random_index = random.choice(prompts)
print(f"Model: {random_model}; prompt: {random_index['inputs']}")

# Generate wallpaper by random prompt
data = query(random_index)

File: generate.ts
import { Buffer } from 'node:buffer'
import { writeFileSync } from 'node:fs'
import { env } from 'node:process'

const API_TOKEN = env.HF_API_TOKEN

async function query(data: object) {
  const response = await fetch(
    'https://api-inference.huggingface.co/models/runwayml/stable-diffusion-v1-5',
    {
      headers: { Authorization: `Bearer ${API_TOKEN}` },
      method: 'POST',
      body: JSON.stringify(data),
    }
  )
  if (!response.ok) {
    throw new Error(`Failed to fetch image: ${response.statusText}`)
  }

  // Return arrayBuffer of blob
  const result = await response.arrayBuffer()
  return result
}

query({
  inputs: 'A pristine, turquoise glacier lake nestled among towering alpine peaks',
}).then(async (response) => {
  const destinationPath = './assets/wallpaper.jpg'
  // create buffer from response
  const buffer = Buffer.from(response)
  // Save image to a local file
  await writeFileSync(destinationPath, buffer)
  console.log(`Image saved to ${destinationPath}`)
})
